lib/Animation.js

```diff
@@ -206,20 +197,19 @@ export class MovingStamp extends Stamp {
     }
 
     /**
-     * Advances the animation to the next frame. If the current frame is the last frame
-     * and the animation is set to loop, it resets to the first frame.
-     *
-     * @returns {number} The updated frame index after progression.
+     * Advances the animation timer. If the timer is greater than the animation duration
+     * and the animation is not looping, playing will be set to false and the timer will
+     * be set to 0.
      */
-    progressFrame() {
-        const FINAL_FRAME_INDEX = this.frames.length - 1;
-        if (this.#currentFrame < FINAL_FRAME_INDEX) {
-            this.#currentFrame += 1;
-        }
-        if (this.#currentFrame === FINAL_FRAME_INDEX && this.#looping) {
-            this.#currentFrame = 0;
+    #progressFrame() {
+        this.#timer += $.time.msElapsed;
```

Comment:
I see your confusion before about time now sorry!

We'd use this.#pen.time to access it rather then $.time

Reply:
Just pushed the change.